### ISSUE TITLE
Shift+Ins performing Copy instead of Paste

### ISSUE THREAD

**Issue by pdanpdan** [microsoft/vscode#88683]
Issue Type: <b>Bug</b>

How to reproduce:

1. Open new empty document and fill with
```
123
456
789
```
2. Select first row (123) and press <kbd>Ctrl</kbd>+<kbd>Insert</kbd>
3. Go on empty row (4) and press <kbd>Shift</kbd>+<kbd>Insert</kbd> - it will paste 123 (GOOD)
4. Select second row (456) and press <kbd>Shift</kbd>+<kbd>Insert</kbd> - it should paste 123 but it does nothing visible except collapsing the selection to the end
5. Go on empty row (5) and press <kbd>Shift</kbd>+<kbd>Insert</kbd> - it will paste 456 (in step 4 it performed copy instead of paste)

Using <kbd>Ctrl</kbd>+<kbd>V</kbd> instead of <kbd>Shift</kbd>+<kbd>Insert</kbd> works as expected.

Might be related to #73462

VS Code version: Code - Insiders 1.42.0-insider (33c79d5ad447956814a2a3658029dffb9e28bae6, 2020-01-15T08:55:46.564Z)
OS version: Linux x64 5.3.0-7625-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz (8 x 3595)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: disabled_software<br>skia_renderer: disabled_off_ok<br>video_decode: unavailable_off<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 1, 1|
|Memory (System)|31.21GB (11.79GB free)|
|Process Argv|--no-sandbox --unity-launch|
|Screen Reader|no|
|VM|0%|
</details>Extensions: none

**Comment by gjsjohnmurray** [comment]
/confirmationPending I followed your instructions but didn't see the problem you reported. Please check your keybindings in case you have  <kbd>Shift</kbd>+<kbd>Insert</kbd> bound to something else as well as to Paste.

**Comment by gjsjohnmurray** [comment]
Hmm, maybe #36170 is significant. The associated [PR](https://github.com/microsoft/vscode/pull/63374) seems to have landed in today's Insiders.

**Comment by pdanpdan** [comment]
Yes, you are right, it's that PR. Just that the primary clipboard has always the current selection if you want to replace some existing text.

I'll re-map the key, but I would let this issue open because I think it is malfunctioning.

**Comment by alexdima** [comment]
@pdanpdan 
I think this is just how selection clipboard works normally...

Every time something is selected, the selection clipboard is set to what is selected. So selecting something and pressing Shift+Insert means that first the selected text is placed in the selection clipboard and then the very same text is pasted on top, effectively causing no change.

So, in step 4. above, selecting 456 is what causes the selection clipboard to become 456.

There is a single setting at `editor.selectionClipboard` which removes support for this, but I believe this is working as expected given the descriptions I've gotten over the years from linux users. 

In other words, I believe selection clipboard cannot be used to overwrite a selection with a previous selection. Do you have an example of an editor that behaves differently around selection clipboard?

**Comment by pdanpdan** [comment]
I fixed my problem by remapping the current <kbd>Shift</kbd>+<kbd>Insert</kbd> to something else.
What I want to point out is that now <kbd>Shift</kbd>+<kbd>Insert</kbd> is pointless if you want to override the current selection with the previously copied text.
Maybe the `When` condition on the keyboard shortcut should include `and text is not selected`?

**Comment by alexdima** [comment]
@pdanpdan How does the default text editor that ships with your linux distro behave in this case?

**Comment by pdanpdan** [comment]
The default editor only uses the mouse selection clipboard when invoked by middle-mouse-button.
But the `editor.selectionClipboard` is perfect. I wasn't aware it's present on linux. Thank you.
